velox/exec/tests/SharedArbitratorTest.cpp

```diff
@@ -831,6 +831,105 @@ DEBUG_ONLY_TEST_F(SharedArbitrationTest, reclaimFromAggregation) {
   }
 }
 
+DEBUG_ONLY_TEST_F(SharedArbitrationTest, reclaimFromDistinctAggregation) {
+  const int numVectors = 32;
+  std::vector<RowVectorPtr> vectors;
+  for (int i = 0; i < numVectors; ++i) {
+    vectors.push_back(newVector());
+  }
+  createDuckDbTable(vectors);
+  std::vector<bool> sameQueries = {false, true};
+  for (bool sameQuery : sameQueries) {
+    SCOPED_TRACE(fmt::format("sameQuery {}", sameQuery));
+    const auto spillDirectory = exec::test::TempDirectoryPath::create();
+    std::shared_ptr<core::QueryCtx> fakeMemoryQueryCtx =
+        newQueryCtx(kMemoryCapacity);
+    std::shared_ptr<core::QueryCtx> aggregationQueryCtx;
+    if (sameQuery) {
+      aggregationQueryCtx = fakeMemoryQueryCtx;
+    } else {
+      aggregationQueryCtx = newQueryCtx(kMemoryCapacity);
+    }
+
+    folly::EventCount fakeAllocationWait;
+    auto fakeAllocationWaitKey = fakeAllocationWait.prepareWait();
+    folly::EventCount taskPauseWait;
+    auto taskPauseWaitKey = taskPauseWait.prepareWait();
+
+    const auto aggregationMemoryUsage = 32L << 20;
+    const auto fakeAllocationSize =
+        kMemoryCapacity - aggregationMemoryUsage + 1;
+
+    std::atomic<bool> injectAllocationOnce{true};
```

Comment:
This test is long. What does it do? How similar is this to other tests? Is there a way to refactor common logic to make tests easier to read?

Reply:
I will think about this in followup to see if there is common pattern that share between similar test cases. Thanks!

Reply:
Ok, but would you explain what does this test do? Somehow I can't figure it out by reading the code.

Reply:
I simplified the test now. Thanks!